src/eva/eva_interactive.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+
+# (C) Copyright 2023 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+import os
+import netCDF4 as nc
+import xarray as xr
+import re
+import numpy as np
+
+from eva.data.data_collections import DataCollections
+from eva.utilities.logger import Logger
+from eva.utilities.timing import Timing
+from eva.data.eva_dataset_base import EvaDatasetFactory
+from eva.transforms.arithmetic import arithmetic, generate_arithmetic_config
+from eva.transforms.accept_where import accept_where, generate_accept_where_config
+
+import eva.plotting.hvplot.interactive_plot_tools as plot
+# --------------------------------------------------------------------------------------------------
+
+
+class EvaInteractive():
+
+    def __init__(self):
+
+        self.logger = Logger('EvaInteractive')
+        self.timer = Timing()
+        self.dc_dict = {}
+        self.fn_dict = {}
+        self.ch_required_dict = {}
+        self.var_cache = []
+
+    # --------------------------------------------------------------------------------------------------
+
+    def load_collection(self, collection_name, filenames, eva_class_name, control_file=None):
+
+        # Handle filenames input
+        if isinstance(filenames, str):
+            filenames = [filenames]
+
+        creator = EvaDatasetFactory()
+        data_collection = DataCollections()
+        eva_object = creator.create_eva_object(eva_class_name, 'data', self.logger, self.timer)
+        config = eva_object.generate_default_config(filenames, collection_name)
+        print(config)
+        eva_object.execute(config, data_collection, self.timer)
+
+        self.dc_dict[collection_name] = data_collection
+        self.fn_dict[collection_name] = filenames[0]
+
+        # Open up file to find channel requirements
+        if eva_class_name is not 'JediLog':
```

Comment:
I think this also needs to be for `mon_data_space` but we don't need to support that datatype here if you don't want

Reply:
Definitely will want to support mon_data_space in interactive. I'll make a note of this. Right now MonDataSpace has an empty generate_default_config method because I wasn't sure how to populate the config for it.

Reply:
@EdwardSafford-NOAA can you help Akira with what the config for MonDataSpace should contain?

Reply:
Sure, I'd be glad to help.  Does that need to happen now (in this PR) or as a follow-up item?  I'll have to study the code a bit to understand what the config should include for `MonDataSpace`.

Reply:
As a first cut I'd include these items in the default_config:
```
'satellite' : None,
'sensor': None,
'control_file': control_file,
'filenames': [ ],
'channels' : [ ],
'regions': [ ],
'levels': [ ],
'groups': [ ],
'name': collection_name
```